packages/server/src/process_move_service.ts

```diff
@@ -0,0 +1,97 @@
+import { Board, Marker, Winner } from "@mapistry/take-home-challenge-shared";
+import { WinningLines } from "./constants";
+
+export class ProcessMoveService {
+    board: Board;
+
+    constructor(board: Board) {
+        this.board = board;
+    }
+
+    static calculateMinimaxScore(board: Board, depth: number, aiTurn: boolean) {
+        const scoreMap = {
+            O: 10,
+            X: -10,
+            tie: 0
+        }
+
+        let finalScore!: number;
+
+        const winner = this.checkForWinner(board)
+        if (winner !== null) return scoreMap[winner]
+        if (this.checkForDraw(board)) return scoreMap.tie
+
+        for (let move = 0; move < board.length; move += 1) {
+            if (board[move] === null) {
```

Comment:
Oh, one thing I noticed was that the styles don't like us using `continue` but that can result in an `if` like this where the entire contents of the `for` loop are nested.

Alternatively, I could have added an early bail like `if (board[move] !== null) continue` and then the rest of the for loop wouldn't be nested.